test/unit/routes/first-time/test-about-you.js

```diff
@@ -1,78 +1,67 @@
+const routeHelper = require('../../../helpers/routes/route-helper')
 const supertest = require('supertest')
 const expect = require('chai').expect
 const proxyquire = require('proxyquire')
 const sinon = require('sinon')
 require('sinon-bluebird')
-const express = require('express')
-const bodyParser = require('body-parser')
-const mockViewEngine = require('../mock-view-engine')
+const ValidationError = require('../../../../app/services/errors/validation-error')
 
+var urlPathValidatorStub
 var stubInsertVisitor
-var stubAboutYouValidator
-var request
+var stubAboutYou
+var app
 
 describe('routes/first-time/about-you', function () {
-  var urlValidatorCalled = false
-
   beforeEach(function () {
+    urlPathValidatorStub = sinon.stub()
     stubInsertVisitor = sinon.stub()
-    stubAboutYouValidator = sinon.stub()
+    stubAboutYou = sinon.stub()
     var route = proxyquire('../../../../app/routes/first-time/about-you', {
       '../../services/data/insert-visitor': stubInsertVisitor,
-      '../../services/validators/first-time/about-you-validator': stubAboutYouValidator,
-      '../../services/validators/url-path-validator': function () { urlValidatorCalled = true }
+      '../../services/domain/about-you': stubAboutYou,
+      '../../services/validators/url-path-validator': urlPathValidatorStub
     })
 
-    var app = express()
-    app.use(bodyParser.urlencoded({ extended: false }))
-    mockViewEngine(app, '../../../app/views')
-    route(app)
-    request = supertest(app)
-    urlValidatorCalled = false
+    app = routeHelper.buildApp(route)
   })
 
   describe('GET /first-time/:dob/:relationship/:benefit/:reference', function () {
-    it('should respond with a 200 for valid path parameters', function (done) {
-      request
+    it('should respond with a 200 for valid path parameters', function () {
+      return supertest(app)
         .get('/first-time/1980-01-01/partner/income-support/1234567')
         .expect(200)
-        .end(function (error, response) {
-          expect(error).to.be.null
-          expect(urlValidatorCalled).to.be.true
-          done()
+        .expect(function () {
+          sinon.assert.calledOnce(urlPathValidatorStub)
```

Comment:
Our routes are constants. We use the same GET and POST routes per route file, so it's possible to replace all instances of the route being written inline with a single const. Makes future changes simple and I think makes the test more clear, with less clutter. 

So: 

	describe('GET /first-time/:dob/:relationship/:benefit/:reference', function () {
		it('should respond with a 200 for valid path parameters', function () {
		  return supertest(app)
		    .get('/first-time/1980-01-01/partner/income-support/1234567')
		    .expect(200)
		    .expect(function () {
		      sinon.assert.calledOnce(urlPathValidatorStub)
		    })
		})
	})

Becomes: 

	const ROUTE = '/first-time/1980-01-01/partner/income-support/1234567'

	describe(`GET ${ROUTE}`, function () {
		it('should respond with a 200 for valid path parameters', function () {
		  return supertest(app)
		    .get(ROUTE)
		    .expect(200)
		    .expect(function () {
		      sinon.assert.calledOnce(urlPathValidatorStub)
		    })
		})
	})

Where we need access to parts of the path for our tests we would write the above const as follows:

	const REFERENCE = '1234567'
	const ROUTE = `/first-time/1980-01-01/partner/income-support/${REFERENCE}`